src/protocols/bdx/tests/TestBdxTransferSession.cpp

```diff
@@ -1,81 +1,296 @@
+#include <protocols/Protocols.h>
 #include <protocols/bdx/BdxMessages.h>
 #include <protocols/bdx/BdxTransferSession.h>
 
 #include <nlunit-test.h>
 
+#include <core/CHIPTLV.h>
 #include <support/CodeUtils.h>
 #include <support/UnitTestRegistration.h>
+#include <system/SystemPacketBuffer.h>
 
 using namespace ::chip;
-using namespace ::chip::BDX;
+using namespace ::chip::bdx;
 
-void SendAndVerifyQuery(nlTestSuite * inSuite, void * inContext, TransferSession & sender, TransferSession & receiver,
+// Use this as a timestamp if not needing to test BDX timeouts.
+const uint64_t kNoAdvanceTime = 0;
+
+// Helper method for generating a complete TLV structure with just a single tag and string
+CHIP_ERROR WriteTLVString(uint8_t * buf, uint32_t bufLen, const char * data, uint64_t tag, uint32_t & written)
```

Comment:
We will be updating the metadata field to require a top level container ("list") where all items have non-anonymous tags, so that metadata can be cleanly parsed. Suggest you test using that format, and have a test TLV validator that ensures nothing comes out that would not be a list (at least).

Reply:
Added restriction for top-level list in test metadata